src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -1936,61 +1950,152 @@ namespace FanCoilUnits {
 			//    Node(FanCoil(FanCoilNum)%HotControlNode)%MassFlowRate = 0.0
 			mdot = 0.0;
 			SetComponentFlowRate( mdot, FanCoil( FanCoilNum ).ColdControlNode, FanCoil( FanCoilNum ).ColdPlantOutletNode, FanCoil( FanCoilNum ).CWLoopNum, FanCoil( FanCoilNum ).CWLoopSide, FanCoil( FanCoilNum ).CWBranchNum, FanCoil( FanCoilNum ).CWCompNum );
-
+			if ( PlantLoop( FanCoil( FanCoilNum ).CWLoopNum ).LoopSide( FanCoil( FanCoilNum ).CWLoopSide ).FlowLock == FlowLocked ) {
+				ColdFlowLocked = true; // check for flow lock
+			}
 			if ( FanCoil( FanCoilNum ).HCoilType_Num == HCoil_Water ) {
 				mdot = 0.0;
 				SetComponentFlowRate( mdot, FanCoil( FanCoilNum ).HotControlNode, FanCoil( FanCoilNum ).HotPlantOutletNode, FanCoil( FanCoilNum ).HWLoopNum, FanCoil( FanCoilNum ).HWLoopSide, FanCoil( FanCoilNum ).HWBranchNum, FanCoil( FanCoilNum ).HWCompNum );
-				// obtain unit output with no active heating/cooling
+				if ( PlantLoop( FanCoil( FanCoilNum ).HWLoopNum ).LoopSide( FanCoil( FanCoilNum ).HWLoopSide ).FlowLock == FlowLocked ) {
+					HotFlowLocked = true; // save locked flow
+				}
 			}
+			// obtain unit output with no active heating/cooling
 			Calc4PipeFanCoil( FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOutNoHC, 0.0 );
-
+			if ( FirstHVACIteration ) {
+				FanCoil( FanCoilNum ).QUnitOutNoHC = QUnitOutNoHC;
+			}
+			else {
+				QUnitOutNoHC = FanCoil( FanCoilNum ).QUnitOutNoHC;
+			}
```

Comment:
If your going to do this, why not wrap all the way up to 1951 and not do those for !FirstHVACIteration.

    if ( FirstHVACIteration ) {
      line 1951 down to 1964 inserted 
      FanCoil( FanCoilNum ).QUnitOutNoHC = QUnitOutNoHC;
     } else {
      QUnitOutNoHC = FanCoil( FanCoilNum ).QUnitOutNoHC;
     }